pitch.md

```diff
@@ -23,38 +89,174 @@ should be more specific than "The user describes properties of unicorns.", e.g.,
+  busy 9am - 11am
+  busy 12pm - 12:30pm
+  busy 3pm - 4pm
+
+On Thursday:
+  free one_hour_of(2pm - 5pm)
+  busy after(7pm)
+
+On Weekdays:
+  free noon-1pm
+  free 2pm - 4pm, badness=**
+  free 6pm - 7pm, badness=***
+
+```
+
+We can see several things about the syntax.  First: it is almost certainly
+destined to be an external DSL, because of its freedom of characters and
+complicated parsing (sometimes times have minutes attached, modifiers for 
+the time of day, potentially military time, and the use of unary asterisk
+counters to represent how undesirable that block would be).
+
+There will also be a file provided by the leader which controls the
+portion of the calendar year which the scheduling could potentially occur.
 
 ## Operation (semantics)
 _What might happen when a program runs? Is there a computational model that
 corresponds with your domain? How does a program interact with the
 user? What kinds of semantic (i.e., non-syntax) errors might occur, and how 
 might they be communicated to the user?_
 
+When a program runs, there are several steps taken. The first step is,
+for each input file, the program parses the provided information into the
+language's internal representation.
+
+The next step in the run of the program is to combine the internal
+representation data generated for each individual into a version of the
+internal representation which contains, for each possible time in the range
+of possible schedules.
+
+Finally, the program would use this internal representation to attempt to
+find a solution to the scheduling problem and output this data to the
+user (perhaps with an explanation of why it was selected and some of the
+second choices -- providing these extra bits of information isn't essential,
+these are more of stretch goals).
+
+One of the semantic troubles which could occur would be if a user put in
+rules which are inconsistent.  One simple example of this might not be
+an error at all. For example, if someone says they are available on Weekdays
+from 1pm to 6pm and then they say that they are busy from 4 to 5 on Tuesdays,
+we don't assume they are inconsistent, we rather assume that both statements
+are valid except for the block during which they are in conflict, for which
+we consider the more specific case.
+
+This is similar to how browsers will interpret CSS, some conflicts are allowed
+and the more specific directive is followed. I would anticipate trying to
+follow this behavior (though probably logging these incidents to the user could
+ensure they were intended).
+
+There is also the opportunity for conflicts with the same specificity, in which
+the best choice seems to be simply returning an error and reporting it to the
+user.
+
+There is also the semantic issue which could result if the combination of schedules
+finds no valid schedule. I'm not sure I would classify this as an error exactly,
+but certainly an unfortunate outcome which the code should be prepared to handle.
```

Comment:
Some constraint problems are unsatisfiable 😞 However, could you possibly present to your user several possibilities with varying degrees of constraint removals to let them pick which one is best. For instance, let's say Cecil is busy every day from 6am - 10pm, and this is causing the constraint problem to be unsatisfiable; perhaps your program could highlight the greatest<sup>1</sup> source of the cause.

<sup>1</sup> I'm using this intentionally vaguely.